subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -662,18 +662,16 @@ static int socket_send_message(struct lwm2m_ctx *client_ctx)
 
 	if (rc < 0) {
 		LOG_ERR("Failed to send packet, err %d", errno);
-		if (msg->type != COAP_TYPE_CON) {
-			lwm2m_reset_message(msg, true);
-		}
-
-		return -errno;
+		rc = -errno;
+	} else {
+		rc = 0;
```

Comment:
Why not return number of bytes sent to the caller?

Reply:
Right.
There was no real reason. I left it to return bytes send.